### PR TITLE
[TEST] Fix TransportReplicationActionTests.testRetryOnReplica for replica request

### PR DESCRIPTION
**Pull request by dakrone**
We were improperly testing that it was a `ConcreteShardRequest` instead of a
`ConcreteReplicaRequest`. This adds that change and also ensures that the
checkpoint is retrievable from the request.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java [hunk 1]
```diff
@@ -911,7 +912,9 @@ protected ReplicaResult shardOperationOnReplica(Request request, IndexShard repl
         assertThat(capturedRequests.size(), equalTo(1));
         final CapturingTransport.CapturedRequest capturedRequest = capturedRequests.get(0);
         assertThat(capturedRequest.action, equalTo("testActionWithExceptions[r]"));
-        assertThat(capturedRequest.request, instanceOf(TransportReplicationAction.ConcreteShardRequest.class));
+        assertThat(capturedRequest.request, instanceOf(TransportReplicationAction.ConcreteReplicaRequest.class));
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java]: I'm not sure if this is enough, this wouldn't reproduce the class cast exception that revealed the issue. Ideally, we would want a test that did *not* have to be changed (other than making it compile) as part of inlining global checkpoints to catch the issue.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Let's chat about this.

**Comment by jasontedor** [comment]
> Let's chat about this.

I discussed this with @dakrone. Changing this particular test is a baby step towards adding a test that would have uncovered the issue that arose with the inlining global checkpoints PR. That will happen in a separate PR.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by dakrone** [comment]
Thanks @jasontedor!

**Comment by jasontedor** [comment]
Thank you @dakrone.

**Pull request merged**